PractiseMay11/CalculatorTask/Calculator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PractiseMay18.CalculatorTask
+{
+    public class Calculator
+    {
+        private double firstNum;
+
+        private double secondNum;
+
+        public Operation Sum { get; set; }
+
+        public Operation Substract { get; set; }
+
+        public Operation Multiply { get; set; }
+
+        public Operation Divide { get; set; }
+
+        public Calculator(double firstNum, double secondNum)
+        {
+            this.firstNum = firstNum;
+            this.secondNum = secondNum;
+            this.Sum = new SumOperation();
+            this.Substract = new SubstractOperation();
+            this.Multiply = new MultiplyOperation();
+            this.Divide = new DivideOperation();
+        }
+
+        public double PerformCalculation(Operation operation, double firstNum, double secondNum)
+        {
+            if (operation == null)
+                throw new ArgumentNullException();
+            if (operation == Divide && secondNum == 0.0)
```

Comment:
Incorrect equality.
In that case we compare references and they will be different.